cryodrgn/analysis.py

```diff
@@ -92,14 +96,15 @@ def cluster_kmeans(z, K, on_data=True, reorder=True):
     labels = kmeans.fit_predict(z)
     centers = kmeans.cluster_centers_
 
+    centers_ind = None
     if on_data:
         centers, centers_ind = get_nearest_point(z, centers)
 
     if reorder:
         g = sns.clustermap(centers)
         reordered = g.dendrogram_row.reordered_ind
         centers = centers[reordered]
-        if on_data:
+        if centers_ind is not None:
```

Comment:
Yeah, this type of refactor is pretty classic pyright and I think is an improvement (?)

Reply:
Yes - in this case what's confusing to the type-checker is doubly confusing to a human reader (variables springing out of existence in certain conditions, only to be used much later), so I think this is a good refactor.